spec.emu

```diff
@@ -384,19 +394,22 @@ location: https://github.com/tc39/proposal-measure/
       1. Let _O_ be the *this* value.
       1. Perform ? RequireInternalSlot(_O_, [[Value]]).
       1. Let _options_ be ? GetOptionsObject(_options_).
-      1. If ? HasOwnProperty(_options_, *"currency"*), throw a *TypeError* exception.
-      1. If ? HasOwnProperty(_options_, *"unit"*), throw a *TypeError* exception.
+      1. Let _unit_ be ? GetOption(_options_, *"unit"*, ~string~, ~empty~, *undefined*).
+      1. Let _currency_ be ? GetOption(_options_, *"currency"*, ~string~, ~empty~, *undefined*).
       1. Let _fractionDigits_ be ? GetOption(_options_, *"fractionDigits"*, ~number~, ~empty~, *undefined*).
       1. Let _significantDigits_ be ? GetOption(_options_, *"significantDigits"*, ~number~, ~empty~, *undefined*).
-      1. If both _fractionDigits_ and _significantDigits_ are not *undefined*, throw a *RangeError* exception.
-      1. If ℝ(_significantDigits_) is not an integer, throw a *RangeError* exception.
-      1. If ℝ(_significantDigits_) < 1, throw a *RangeError* exception.
-      1. If ℝ(_fractionDigits_) is not an integer, throw a *RangeError* exception.
-      1. If ℝ(_fractionDigits_) < 0, throw a *RangeError* exception.
+      1. If _currency_ is not *undefined* and _unit_ is not *undefined*, throw a *RangeError* exception.
```

Comment:
Also need to throw if either _currency_ or _unit_ is defined, and _O_.[[Currency]] or _O_.[[Unit]] is defined.

They also need to be validated as in the constructor, so it's probably best to define an abstract operation like GetAmountOptions that's used by both the constructor and `with()` to parse the options.

Reply:
Should be done now. I've added a new AO as suggested.